pkg/apis/serving/k8s_validation.go

```diff
@@ -416,6 +416,27 @@ func validateContainerPorts(ports []corev1.ContainerPort) *apis.FieldError {
 	return errs
 }
 
+func validateReadinessProbe(p *corev1.Probe) *apis.FieldError {
+	if p == nil {
+		return nil
+	}
+
+	errs := validateProbe(p)
+
```

Comment:
Looks like we have previously not validated the bounds of periodSeconds and let the K8s API server handle the validation. Since we are using the periodSeconds in our controllers now we probably should add the validation that this is a positive number.

Reply:
I'd probably also split the changes to `pkg/apis` into it's own change, since it's fairly independent.

Reply:
Added a check to make sure periodSeconds is non-negative (allowing 0 to signal the aggressive probe) and checks to make sure timeoutSeconds and failureThreshold are non-negative in the case where we're not using the aggressive probe.